sw/device/tests/sim_dv/pwrmgr_deep_sleep_all_wake_ups.c

```diff
@@ -0,0 +1,316 @@
+      .counter_threshold = 0 /* Don't need for posedge detection */,
+  };
+  CHECK_DIF_OK(dif_pinmux_wakeup_detector_enable(dif, kPinmuxWkupDetector5,
+                                                 detector_cfg));
+}
+
+static const test_wakeup_sources_t kTestWakeupSources[] = {
+    {
+        .name = "SYSRST_CTRL",
+        .base_addr = TOP_EARLGREY_SYSRST_CTRL_AON_BASE_ADDR,
+        .dif_handle = &sysrst_ctrl,
+        .init = sysrst_ctrl_init,
+        .wakeup_src = kDifPwrmgrWakeupRequestSourceOne,
+        .config = prgm_sysrst_ctrl_wakeup,
+    },
+    {
+        .name = "ADC_CTRL",
+        .base_addr = TOP_EARLGREY_ADC_CTRL_AON_BASE_ADDR,
+        .dif_handle = &adc_ctrl,
+        .init = adc_ctrl_init,
+        .wakeup_src = kDifPwrmgrWakeupRequestSourceTwo,
+        .config = prgm_adc_ctrl_wakeup,
+    },
+    {
+        .name = "PINMUX",
+        .base_addr = TOP_EARLGREY_PINMUX_AON_BASE_ADDR,
+        .dif_handle = &pinmux,
+        .init = pinmux_init,
+        .wakeup_src = kDifPwrmgrWakeupRequestSourceThree,
+        .config = prgm_pinmux_wakeup,
+    },
+    {
+        .name = "PWRMGR",
+        .base_addr = TOP_EARLGREY_PWRMGR_AON_BASE_ADDR,
+        .dif_handle = &pwrmgr,
+        .init = pwrmgr_init,
+        .wakeup_src = (dif_pwrmgr_request_sources_t)kNoWkupDev,
+    },
+    {
+        .name = "RV_PLIC",
+        .base_addr = TOP_EARLGREY_RV_PLIC_BASE_ADDR,
+        .dif_handle = &rv_plic,
+        .init = rv_plic_init,
+        .wakeup_src = (dif_pwrmgr_request_sources_t)kNoWkupDev,
+    },
+};
+
+/**
+ * External interrupt handler.
+ */
+void ottf_external_isr(void) {
+  dif_pwrmgr_irq_t irq_id;
+  top_earlgrey_plic_peripheral_t peripheral;
+
+  isr_testutils_pwrmgr_isr(plic_ctx, pwrmgr_isr_ctx, &peripheral, &irq_id);
+
+  // Check that both the peripheral and the irq id is correct
+  CHECK(peripheral == kTopEarlgreyPlicPeripheralPwrmgrAon,
+        "IRQ peripheral: %d is incorrect", peripheral);
+  CHECK(irq_id == kDifPwrmgrIrqWakeup, "IRQ ID: %d is incorrect", irq_id);
+}
+
+static void execute_test(int wakeup_source) {
+  // configure wakeup device
+  kTestWakeupSources[round].config(kTestWakeupSources[round].dif_handle);
+  // Deep sleep
+  pwrmgr_testutils_enable_low_power(&pwrmgr,
+                                    kTestWakeupSources[round].wakeup_src, 0);
+  LOG_INFO("Issue WFI to enter sleep %d", round);
+  wait_for_interrupt();
+}
+
+/**
+ * Clean up pwrmgr wakeup reason register for the next round
+ */
+static void delay_n_clear(uint32_t delay_in_us) {
+  busy_spin_micros(delay_in_us);
+  CHECK_DIF_OK(dif_pwrmgr_wakeup_reason_clear(&pwrmgr));
+}
+
+bool test_main(void) {
+  // Enable global and external IRQ at Ibex.
+  irq_global_ctrl(true);
+  irq_external_ctrl(true);
+
+  // device init
+  for (size_t i = 0; i < ARRAYSIZE(kTestWakeupSources); ++i) {
+    kTestWakeupSources[i].init(kTestWakeupSources[i].dif_handle,
+                               kTestWakeupSources[i].base_addr);
+  }
```

Comment:
Sorry what I meant was, just call the Init DIFs directly here, and remove the init function pointer from your `test_wakeup_sources_t` struct. (you can keep the config function pointer), then you don't need the wrappers at all, and it would be about the same LOC as the original setup, where you had a macro, func pointers, AND this loop.
i.e., replace this loop with:
```c
CHECK_DIF_OK(dif_sysrst_ctrl_init(...));
CHECK_DIF_OK(dif_adc_ctrl_init(...));
CHECK_DIF_OK(dif_pinmux_init(...));
...
```

Reply:
fixed.